sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuMultiFileReader.scala

```diff
@@ -280,6 +336,110 @@ object MultiFileReaderUtils {
       partsCols
     }
   }
+
+  private def buildPartitionsColumns(partRowNumsAndValues: Array[(Long, InternalRow)],
+      partSchema: StructType): Array[GpuColumnVector] = {
+    // build the partitions vectors for all partitions within each column
+    // and concatenate those together then go to the next column
+    closeOnExcept(new Array[GpuColumnVector](partSchema.length)) { partsCols =>
+      for ((field, colIndex) <- partSchema.zipWithIndex) {
+        val dataType = field.dataType
+        withResource(new Array[ColumnVector](partRowNumsAndValues.length)) { onePartCols =>
+          partRowNumsAndValues.zipWithIndex.foreach { case ((rowNum, valueRow), partId) =>
+            val singleValue = valueRow.get(colIndex, dataType)
+            withResource(GpuScalar.from(singleValue, dataType)) { oneScalar =>
+              onePartCols(partId) = ColumnVector.fromScalar(oneScalar, rowNum.toInt)
+            }
+          }
+          val res = if(onePartCols.length > 1) {
+            ColumnVector.concatenate(onePartCols: _*)
+          } else {
+            onePartCols.head
+          }
+          partsCols(colIndex) = GpuColumnVector.from(res, field.dataType)
+        }
+      }
+      partsCols
+    }
+  }
+
+  /**
+   * Splits the (partRows, partValues) into multiple batches such that sum
+   * of sizes does not exceed cudf limit
+   *
+   * Example,
+   * Input:
+   *  partRows:   [ 10, 40, 70, 10, 11 ]
+   *  partValues: [ [abc, ab], [bc, ab], [abc, bc], [aa, cc], [ade, fd] ]
+   *
+   * Assume:
+   *  limit:      300 bytes
+   *
+   * Split Calculation:
+   *  At rowIndex = 2:
+   *  -> rollingSum(colIndex=0) = (10 * 3) + (40 * 2) = 110
+   *  -> rollingSum(colIndex=1) = (10 * 2) + (40 * 2) = 100
+   *  -> rollingSum(0) + (70 * 3) > 300
```

Comment:
sorry this is confusing to me, I'm not sure what rollingSum(0) is.  

Perhaps calculate the size of the first partition Rows.. column 0, 1 at 10 rows = 50 + next set of rows = 160 = 210 but adding third setof 70*3 + 70 *2 = 350 is > 300...  And really that condition by itself isn't made clear in the description here.. that by itself is == 350 so assumption is that gets truncated and made into smaller batches.

Reply:
Updated documentation with example